prover/zkevm/arithmetization/arithmetization.go

```diff
@@ -18,86 +18,57 @@ import (
 // Settings specifies the parameters for the arithmetization part of the zkEVM.
 type Settings struct {
 	Limits *config.TracesLimits
-	// IgnoreCompatibilityCheck disables the strong compatibility check.
-	// Specifically, it does not require the constraints and the trace file to
-	// have both originated from the same commit.  By default, the compability
-	// check should be enabled.
 	IgnoreCompatibilityCheck *bool
-	// OptimisationLevel determines the optimisation level which go-corset will
-	// apply when compiling the zkevm.bin file to AIR constraints.  If in doubt,
-	// use mir.DEFAULT_OPTIMISATION_LEVEL.
-	OptimisationLevel *mir.OptimisationConfig
+	OptimisationLevel        *mir.OptimisationConfig
 }
 
 // Arithmetization exposes all the methods relevant for the user to interact
-// with the arithmetization of the zkEVM. It is a sub-component of the whole
-// ZkEvm object as it does not includes the precompiles, the keccaks and the
-// signature verification.
+// with the arithmetization of the zkEVM.
 type Arithmetization struct {
 	Settings *Settings
-	// Schema defines the columns, constraints and computations used to expand a
-	// given trace, and to subsequently to check satisfiability.
-	Schema *air.Schema
-	// Metadata embedded in the zkevm.bin file, as needed to check
-	// compatibility.  Guaranteed non-nil.
+	Schema   *air.Schema
 	Metadata typed.Map
 }
 
-// NewArithmetization is the function that declares all the columns and the constraints of
-// the zkEVM in the input builder object.
 func NewArithmetization(builder *wizard.Builder, settings Settings) *Arithmetization {
 	schema, metadata, errS := ReadZkevmBin(settings.OptimisationLevel)
 	if errS != nil {
 		panic(errS)
 	}
-
 	Define(builder.CompiledIOP, schema, settings.Limits)
-
 	return &Arithmetization{
 		Schema:   schema,
 		Settings: &settings,
 		Metadata: metadata,
 	}
 }
 
-// Assign the arithmetization related columns. Namely, it will open the file
-// specified in the witness object, call corset and assign the prover runtime
-// columns. As part of the assignment processs, the original trace is expanded
-// according to the given schema.  The expansion process is about filling in
-// computed columns with concrete values, such for determining multiplicative
-// inverses, etc.
 func (a *Arithmetization) Assign(run *wizard.ProverRuntime, traceFile string) {
 	traceF := files.MustRead(traceFile)
-	// Parse trace file and extract raw column data.
 	rawColumns, metadata, errT := ReadLtTraces(traceF, a.Schema)
 
-	// Performs a compatibility check by comparing the constraints
-	// commit of zkevm.bin with the constraints commit of the trace file.
-	// Panics if an incompatibility is detected.
 	if !*a.Settings.IgnoreCompatibilityCheck {
-		var errors []string
+		var compatibilityErrors []string
```

Comment:
Thinking about it, but we should be using an `errors.Join` pattern I think here.